atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/internalschema/persistence/CoordinationServices.java

```diff
@@ -69,6 +80,18 @@ private static CoordinationStore<VersionedInternalSchemaMetadata> createCoordina
                 initializeAsync);
     }
 
+    private static <T> CoordinationStore<T> createTransactionKeyValueServiceManagerCoordinationStore(
+            KeyValueService keyValueService, LongSupplier timestampSupplier, Class<T> clazz, boolean initializeAsync) {
+        return KeyValueServiceCoordinationStore.create(
+                ObjectMappers.newServerObjectMapper(),
+                keyValueService,
+                AtlasDbConstants.DEFAULT_TRANSACTION_KEY_VALUE_SERVICE_COORDINATION_KEY,
+                timestampSupplier,
+                Object::equals,
+                clazz,
```

Comment:
nit: You probably got this from `KeyValueServiceCoordinationStore`, but we should probably name this `valueClass` or similar.

Reply:
I thought that's pretty standard java naming convention for when you pass the type as an argument. But I'll rename

Reply:
Sorry about this, keeping `clazz` is fine - the usual test I use is "is it obvious what the `clazz` is trying to do?" and I guess at the time I reviewed this I didn't think it was, but actually since you return a `CoordinationStore<T>` it should be. My bad.